sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs

```diff
@@ -142,6 +142,21 @@ protected async Task SetAuthorizationHeader(HttpMessage message, TokenRequestCon
             message.Request.Headers.SetValue(HttpHeader.Names.Authorization, headerValue);
         }
 
+        protected async Task<string> GetAuthorizationHeader(HttpMessage message, TokenRequestContext context, bool async)
```

Comment:
@christothes, would adding this method create a security concern for the Core challenge-based auth policy?

Reply:
Can you talk a bit about why this is needed?

Reply:
My thinking was this (although if there's a better way, please tell me...!)...

- I need to add the AAD Access Token[ to the body of a request](https://github.com/Azure/azure-sdk-for-net/pull/19696/files?file-filters%5B%5D=.cs#diff-a20df2296283534f7f3679f94372d8a34fa2b468610dbb5a37e4997f5a008079R85) to get an ACR refresh token back.
- The `BearerTokenChallengeAuthenticationPolicy` is adding this to the `Authorization` request header, but not exposing it elsewhere - the token cache is not visible outside the policy.
- Being able to get this via a protected method lets me add it to the body of my request.

Reply:
Thanks, I'm just trying to make sure I understand the entire auth scenario to see if there is another approach. 

In the `ContainerRegistryCredentialsPolicy` auth flow, what happens on the first request before there is any value in the cache?

Assuming this is not the first request, Is the token that would be in the cache the same one that we previously had set on the `Authorization` header? If so, would it already be available on the `Request` when you are in `AuthenticateRequestOnChallengeAsync`?

Reply:
> In the `ContainerRegistryCredentialsPolicy` auth flow, what happens on the first request before there is any value in the cache?

From my understanding, it looks like when the first request comes through, there is already a value in the cache, because credential and scopes were set when `_accessTokenCache` was [initialized in the constructor](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/Shared/BearerTokenChallengeAuthenticationPolicy.cs#L46).

> Assuming this is not the first request, Is the token that would be in the cache the same one that we previously had set on the `Authorization` header? If so, would it already be available on the `Request` when you are in `AuthenticateRequestOnChallengeAsync`?

Ah, I see.  So the authorization header value is already on the `message.Request` and I could get it from there.  This appears to be the case for both the first request and subsequent requests.

I'm wondering if there should be any concern about timing issues if, say, the cache were to have its value updated after it had set the header on the request and before we retrieved the value from the header -- thoughts on this?

Reply:
> I'm wondering if there should be any concern about timing issues if, say, the cache were to have its value updated after it had set the header on the request and before we retrieved the value from the header -- thoughts on this?

I think the worst case scenario on the timing issue is that you would fail and need to retry again. But in practice the token would most likely still be valid even if stale, since we refresh in the background when the token is less than 5 mins from expiration.